### PR TITLE
bug fix in MIM attack function

### PR DESCRIPTION
**Pull request by ypang6**
Added softmax cross-entropy loss as in the original paper.

### CODE REVIEW

**Comment by ypang6** [comment]
This pull request is raised to address the issue in #1191.

**Comment by alkaet** [comment]
Thanks a lot for PR.
Can you format the code using Black?

**Comment by ypang6** [comment]
> Thanks a lot for PR.
> Can you format the code using Black?

No problem. I was having some issues formatting the code so I edited it manually. Please let me know if there is any other issue.

**Comment by alkaet** [comment]
I'll have another look and test it once it is formatted in Black.

**Comment by alkaet** [comment]
lgtm. Thanks!
